### PR TITLE
Refactor update-dependencies to update the runtime Dockerfiles

### PR DESCRIPTION
**Pull request by MichaelSimons**
I'm not a fan of this.  I view it as a stop-gap until the produces and consumes build tools infrastructure is in place.

### CODE REVIEW

File: update-dependencies/Program.cs [hunk 1]
```diff
@@ -63,25 +65,35 @@ private static bool ParseArgs(string[] args)
...
+
+            IEnumerable<DependencyBuildInfo> buildInfos = new[]
+                {
+                    new DependencyBuildInfo(cliBuildInfo, false, Enumerable.Empty<string>()),
+                    new DependencyBuildInfo(
+                        new BuildInfo() {
+                            Name = SharedFrameworkBuildInfoName,
+                            LatestReleaseVersion = sharedFrameworkVersion,
+                            LatestPackages = new Dictionary<string, string>()
+                            },
```
Comment (dagood) [inline update-dependencies/Program.cs]: \<nit> Unindent?

File: update-dependencies/Program.cs [hunk 2]
```diff
@@ -98,11 +110,12 @@ private static Task CreatePullRequest(DependencyUpdateResults updateResults)
 
         private static IEnumerable<IDependencyUpdater> GetUpdaters()
         {
-            return Directory.GetFiles(s_repoRoot, "Dockerfile", SearchOption.AllDirectories)
-                .Select(path => CreateDependencyUpdater(path));
+            string[] dockerfiles = Directory.GetFiles(s_repoRoot, "Dockerfile", SearchOption.AllDirectories);
+            return dockerfiles.Select(path => CreateSdkUpdater(path))
+                .Union(dockerfiles.Select(path => CreateRuntimeUpdater(path)));
```
Comment (dagood) [inline update-dependencies/Program.cs]: Isn't `Concat` enough?

File: update-dependencies/update-dependencies.csproj [hunk 3]
```diff
@@ -6,7 +6,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.DotNet.VersionTools" Version="1.0.26-prerelease-00708-05" />
+    <PackageReference Include="Microsoft.DotNet.VersionTools" Version="1.0.27-prerelease-01428-01" />
     <PackageReference Include="System.Diagnostics.TextWriterTraceListener" Version="4.0.0" />
+    <PackageReference Include="NuGet.Packaging.Core.Types" Version="4.0.0" />
+    <PackageReference Include="NuGet.Packaging" Version="4.0.0" />
```
Comment (dagood) [inline update-dependencies/update-dependencies.csproj]: Why were the NuGet packages added?

File: update-dependencies/CliDependencyHelper.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        private static readonly Lazy<HttpClient> DownloadClient = new Lazy<HttpClient>();
+
+        public static string GetSharedFrameworkVersion(string cliVersion)
+        {
+            Trace.TraceInformation($"Looking for the Shared Framework CLI '{cliVersion}' depends on.");
+
+            string cliCommitHash = GetCommitHash(cliVersion);
+            XDocument depVersions = DownloadDependencyVersions(cliCommitHash).Result;
+            XNamespace msbuildNamespace = "http://schemas.microsoft.com/developer/msbuild/2003";
```
Comment (dagood) [inline update-dependencies/CliDependencyHelper.cs]: I usually get the namespace of the `XDocument` for flexibility, and I might use `Descendants` below, but it's probably not needed here.

File: update-dependencies/Program.cs (hunk 1 above)
Comment (MichaelSimons) [inline update-dependencies/Program.cs]: Fixed.

File: update-dependencies/Program.cs (hunk 2 above)
Comment (MichaelSimons) [inline update-dependencies/Program.cs]: Yes - changed.

File: update-dependencies/CliDependencyHelper.cs (hunk 4 above)
Comment (MichaelSimons) [inline update-dependencies/CliDependencyHelper.cs]: Good point on the namespace - changed.
It seems more precise to use Elements with an explicit path - leaving.

File: update-dependencies/update-dependencies.csproj (hunk 3 above)
Comment (MichaelSimons) [inline update-dependencies/update-dependencies.csproj]: Picking up the new versions package required 4.0.0 of NuGet.  Without, it is trying to use 3.5.0.  I was able to remove the NuGet.Packaging.Core.Types reference.

Comment (MichaelSimons) [inline update-dependencies/update-dependencies.csproj]: I should reword, without it it is causing a build warning - warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.

Comment (dagood) [inline update-dependencies/update-dependencies.csproj]: Looks like the NuGet.Packaging dependency version in the VersionTools nuspec didn't get updated when the one in the VersionTools project.json did (annoying duplication there)... which means this fix makes sense, but it seems like this should really be fixed in BuildTools.

**Pull request merged**